src/neo/Network/P2P/TaskSession.cs

```diff
@@ -13,6 +13,7 @@ internal class TaskSession
 
         public bool IsFullNode { get; }
         public uint LastBlockIndex { get; set; }
+        public long LastStateIndex { get; set; }
```

Comment:
If we attach the state in the consensus, always will be Block-1, right?

Reply:
In neox-2.x, nodes determine whether to sync state root by comparing its `StateHeight` and `Height`. 
I think we can recover ping payload in neox.

Reply:
Please don't resolve the conversation, otherwise I can't see the answer

Reply:
If we will send the state root with the consensus data, we should store this information in the `BlockHeader`, and have a fixed index (Height-1)